dbt_platform_helper/domain/codebase.py

```diff
@@ -154,31 +157,34 @@ def build(self, app: str, codebase: str, commit: str):
 
         raise ApplicationDeploymentNotTriggered(codebase)
 
-    def deploy(self, app, env, codebase, commit):
+    def deploy(self, app: str, env: str, codebase: str, commit: str = None, ref: str = None):
         """Trigger a CodePipeline pipeline based deployment."""
-        session = self.get_aws_session_or_abort()
 
-        application = self.load_application(app, default_session=session)
-        if not application.environments.get(env):
-            raise ApplicationEnvironmentNotFoundException(application.name, env)
+        self.validate_commit_and_ref_flags(commit, ref)
 
-        self.check_image_exists(session, application, codebase, commit)
+        application, session = self.populate_application_values(app, env)
 
-        codepipeline_client = session.client("codepipeline")
+        image_ref = f"commit-{commit}" if commit else ref
+        image_details = self.get_image_details(session, application, codebase, image_ref)
+        image_ref = self.retrieve_commit_tag(image_ref, image_details)
 
+        codepipeline_client = session.client("codepipeline")
         pipeline_name = self.get_manual_release_pipeline(codepipeline_client, app, codebase)
 
+        confirmation_message = f'You are about to deploy "{app}" for "{codebase}" with image reference "{image_ref}" to the "{env}" environment using the "{pipeline_name}" deployment pipeline. Do you want to continue?'
```

Comment:
Maybe line 167 should be `original_image_ref` then you can have something like:

```suggestion
        corresponding_to = f" (corresponding to {original_image_ref}" if original_image_ref != image_ref else ""
        confirmation_message = f'You are about to deploy "{app}" for "{codebase}" with image reference "{image_ref}"{corresponding_to} to the "{env}" environment using the "{pipeline_name}" deployment pipeline. Do you want to continue?'
```
or similar.